File: test-fixtures/security/vulnerable.java
package com.example.security;

// INTENTIONAL VULNERABILITIES - DO NOT USE IN PRODUCTION
// Test fixture for security scanner validation

import java.io.*;
import java.sql.*;
import javax.xml.parsers.*;

public class VulnerableJava {

    // JAVA-001: SQL Injection via string concatenation
    public void searchUser(String userId) throws SQLException {
        String query = "SELECT * FROM users WHERE id = " + userId; // BAD
        Statement stmt = connection.createStatement();
        stmt.executeQuery(query); // BAD: Executes with concatenated string
    }

    // JAVA-002: XXE Vulnerability
    public void parseXml(InputStream input) throws Exception {
        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance(); // BAD: Default config
        DocumentBuilder builder = factory.newDocumentBuilder();
        builder.parse(input);
    }

    // JAVA-003: Insecure Deserialization
    public Object deserializeData(InputStream stream) throws Exception {
        ObjectInputStream ois = new ObjectInputStream(stream); // BAD
        return ois.readObject(); // BAD: Arbitrary object deserialization
    }

    // JAVA-004: Path Traversal
    public void readFile(HttpServletRequest request) throws IOException {
        String filename = request.getParameter("file");
        File file = new File("/data/" + filename); // BAD: User input in path
        Files.readAllBytes(Paths.get(request.getParameter("path"))); // BAD
    }

    // JAVA-005: LDAP Injection
    public void searchLdap(String username) throws NamingException {
        String filter = "(uid=" + username + ")"; // BAD: String concat
        ctx.search("ou=users,dc=example,dc=com", filter, null); // BAD
    }

    // SAFE PATTERNS (should not trigger)
    public void safeSearch(String userId) throws SQLException {
        PreparedStatement pstmt = connection.prepareStatement(
            "SELECT * FROM users WHERE id = ?");
        pstmt.setString(1, userId); // GOOD: Parameterized
        pstmt.executeQuery();
    }
}

File: test-fixtures/security/vulnerable.cs
// INTENTIONAL VULNERABILITIES - DO NOT USE IN PRODUCTION
// Test fixture for security scanner validation

using System;
using System.Data.SqlClient;
using System.IO;
using System.Runtime.Serialization.Formatters.Binary;
using System.Web.Mvc;

namespace VulnerableCSharp
{
    public class UserController : Controller
    {
        // CSHARP-001: SQL Injection
        public void GetUser(string userId)
        {
            var query = "SELECT * FROM users WHERE id = " + userId; // BAD
            var cmd = new SqlCommand(query); // BAD
            cmd.ExecuteReader(); // BAD
        }

        // CSHARP-002: Insecure Deserialization
        public object DeserializeData(Stream stream)
        {
            var formatter = new BinaryFormatter(); // BAD: BinaryFormatter is insecure
            return formatter.Deserialize(stream); // BAD
        }

        // CSHARP-003: XSS via Razor
        public ActionResult DisplayContent(string content)
        {
            ViewBag.Content = Html.Raw(content); // BAD: Raw HTML
            var html = new HtmlString(content); // BAD
            Response.Write(content); // BAD
            return View();
        }

        // CSHARP-004: Path Traversal
        public void ReadFile()
        {
            var path = Path.Combine("C:\\Data", Request.QueryString["file"]); // BAD
            var content = File.ReadAllText(Request.QueryString["path"]); // BAD
        }

        // CSHARP-005: LDAP Injection
        public void SearchLdap(string username)
        {
            var filter = "(&(objectClass=user)(uid=" + username + "))"; // BAD
            var searcher = new DirectorySearcher();
            searcher.Filter = filter; // BAD
        }

        // SAFE PATTERNS
        public void SafeQuery(string userId)
        {
            var cmd = new SqlCommand("SELECT * FROM users WHERE id = @id");
            cmd.Parameters.AddWithValue("@id", userId); // GOOD: Parameterized
        }
    }
}
